ref(eventstore): Move bind_nodes to eventstore

lynnagara opened the pull request:
Use `eventstore.bind_nodes` instead of `EventManager.bind_nodes` everywhere

fpacifici commented on bin/load-mocks:
```diff
@@ -197,7 +198,7 @@ def create_sample_time_series(event, release=None):
         return
 
-    Event.objects.bind_nodes([event], 'data')
+    eventstore.bind_nodes([event])
```
Now that we start using this method, I think we should take a step back and figure out how it will work for any fields that is not 'data'. 'data' fetches everything from nodestore and now it is only used for that :
1) How will this be called for other fields? Is this actually going to support other fields? How is it going to work for nested fields ?
2) Are we going to expose where the fields are fetched from ? Nodestore vs Snuba
3) Are we going to expose which fields have been fetched and which ones have not ?

fpacifici reviewed (commented):
Nothing specifically against this PR, but before going ahead with bind_nodes, I think we should nail down the longer term plan for that method.

lynnagara committed:
feat: Only bind node data if event timestamp matches

Pretty soon we won't have a guarantee of unique event IDs in Snuba,
since event ids are not deduplicated over different days. Whilst
duplicate event IDs should be an edge case, we want to make sure that
bind_nodes never attaches incorrect data to an event object. bind_nodes
now checks that the timestamp of the bound event matches the data
fetched from nodestore. If they are different, we do not actually bind
the fetched data.
